WcaOnRails/app/javascript/edit-schedule/SchedulesEditor/ActivityPicker.jsx

```diff
@@ -0,0 +1,344 @@
+            );
+          })}
+          <Col xs={12}>
+            <p>
+              Want to add a custom activity such as lunch or registration?
+              Click and select a timeframe on the calendar!
+            </p>
+          </Col>
+        </Panel.Body>
+      </Panel>
+    );
+  }
+}
+
+function ActivityPickerLine({ eventWcif, usedActivityCodeList, selectedLine, selectedX }) {
+  let event = events.byId[eventWcif.id];
+
+  return (
+    <Col xs={12} className="event-picker-line">
+      <Row>
+        <Col xs={12} md={3} lg={2} className="activity-icon">
+          <span className={cn("cubing-icon", `event-${event.id}`)}></span>
+        </Col>
+        <Col xs={12} md={9} lg={10}>
+          <Row>
+            {eventWcif.rounds.map((value, index) => {
+              let activities = (
+                <ActivitiesForRound key={value.id}
+                                    indexInRow={index}
+                                    round={value}
+                                    usedActivityCodeList={usedActivityCodeList}
+                                    selectedLine={selectedLine}
+                                    selectedX={selectedX}
+                />
+              );
+              if (event.id == "333mbf" || event.id == "333fm") {
+                // For these events the selectedX spreads accross multiple rounds.
+                // This corrects the offset.
+                selectedX -= formats.byId[value.format].expectedSolveCount;
+              }
+              return activities;
+            })}
+          </Row>
+        </Col>
+      </Row>
+    </Col>
+  );
+}
+
+function ActivitiesForRound({ usedActivityCodeList, round, selectedLine, selectedX, indexInRow }) {
+  let activityCode = round.id;
+  let { eventId } = parseActivityCode(activityCode);
+
+  let attempts = [];
+  if (eventId == "333fm" || eventId == "333mbf") {
+    let numberOfAttempts = formats.byId[round.format].expectedSolveCount;
+    for (let i = 0; i < numberOfAttempts; i++) {
+      attempts.push(<ActivityForAttempt activityCode={activityCode}
+                                        usedActivityCodeList={usedActivityCodeList}
+                                        key={i}
+                                        attemptNumber={i+1}
+                                        selected={selectedLine && selectedX == i}
+      />);
+    }
+    attempts.push(<div key={numberOfAttempts} className="clearfix" />);
+  } else {
+    attempts.push(<ActivityForAttempt key="0" usedActivityCodeList={usedActivityCodeList}
+                                              activityCode={activityCode}
+                                              selected={selectedLine && selectedX == indexInRow}
+                                              attemptNumber={null}
+                  />);
+  }
+  return (
```

Comment:
Can this just become `return attempts` instead? I think react supports returning array.